### PR TITLE
S&C test across banner, header and epic channels

### PR DESCRIPTION
**Pull request by svillafe**
## TODO before merging
- [x] Add support for liveblogs - SVF
- [x] Debug double-epic issue (possibly connected to A/B test issues in above point)- SVF
- [x] Check if we were able to track one-off contribution via support site.- SVF
- [x] ~~Hide subscribe link in the header's support variant~~
Note 21/8: We decided to drop the header from the test. The traffic is not needed, nor accounted for by Chay. Also, adding the header introduces new issues, eg a weird experience in the header, potential to throw off the results - SVF, AM
- [x] Remove the header changes - SVF
- [x] Update `products` to reflect new paper subscriptions types in #17662)

- [x] Add campaign code to subs header links in control variant.
- [x] Update links for header in support variant across different breakpoints. 
- [x] Update subscription frontend. (https://github.com/guardian/subscriptions-frontend/pull/979)

## TODO in separate PRs, before merging
- [x] ~~Create another set of promo codes for tracking control variant in subs~~ - SVF
Note 21/8: We may not need to do this after deciding to drop the header.
SVF: After checking the different flows we don't need to create another set of promo codes for control since there isn't a channel which links to subscribe platform from control.
- [x] ~~Ensure subs site can translate control campaign code into a promo code~~ - SVF
Note 21/8: We may not need to do this after deciding to drop the header.
- [x] Update Support frontend with support variant promo codes. - SVF  (**PR**: https://github.com/guardian/support-frontend/pull/181)
- [x] Send component events for banner and header links (#17662, https://github.com/guardian/ophan/pull/2379) - JS
- [x] Updated thrift schema (https://github.com/guardian/ophan/pull/2384) and ComponentEvent extractor in the slab (https://github.com/guardian/ophan/pull/2381)

## TODO in separate PRs, at a later date
- [ ] Send component type and component id in URL to contributions, membership and support sites, so they can be stored with the acquisitions events which will go in the new acquisitions table - JS
- [ ] Fix A/B test handling of forced variants via # in URL, so that engagement banner can be tested locally - JS

## What does this change?

This PR adds the logic to handle the different channels for S&C's fourth test. 

### Channels that we are testing:
* Header
* Epic
* Live blog
* Engagement banner.

### Impressions
These can be tracked via the `ab_tests` field on `pageview`. Note that since the header appears on every page, we register an A/B test participation even when neither the banner nor epic was shown. So the number of "impressions" of one of the variants should be very close to the total pageviews for browsers in that variant.

#### Component events
For further analysis, we're also tracking impressions & views of the banner and epic using the `component_events` field which will soon be available on `pageview`. Campaign codes are as follows (code for the epic is the same for liveblog & article but these can be distinguished by looking at the `content_type_tag` in `content_new`):

**support, epic**: gdnwb_copts_memco_sandc_support_baseline_support_epic
**support, banner**: gdnwb_copts_memco_sandc_support_baseline_support_banner

**control, epic**: gdnwb_copts_memco_sandc_support_baseline_control_epic
**control, banner**: gdnwb_copts_memco_sandc_support_baseline_control_banner

####  Control

| Channel | Campaign Code |
|---------|---------------|
|**Header Become Supporter**| gdnwb_copts_memco_sandc_support_baseline_control_header_become_supporter|
|**Header Subscribe**|gdnwb_copts_memco_sandc_support_baseline_control_header_subscribe|
| **Engagement Banner** | gdnwb_copts_memco_sandc_support_baseline_control_banner |
| **Epic**  | gdnwb_copts_memco_sandc_support_baseline_control_epic |
| **Live Blog** | gdnwb_copts_memco_sandc_support_baseline_control_epic_liveblog|

#### Variant Support

| Channel | Campaign Code |
|---------|---------------|
|**Header Become a Supporter** | gdnwb_copts_memco_sandc_support_baseline_support_header_become_supporter |
|**Header Subscribe** | gdnwb_copts_memco_sandc_support_baseline_support_header_subscribe|
| **Engagement Banner** | gdnwb_copts_memco_sandc_support_baseline_support_banner  |
| **Epic**  | gdnwb_copts_memco_sandc_support_baseline_support_epic |
| **Liveblog** | gdnwb_copts_memco_sandc_support_baseline_support_liveblog |

#### Variant Control
|Product | Promo code | 
|---------|------------|
|Digital | DJ6P8DA5J|
|Paper + Digital |NJ6P8BJHF |
|Paper|NJ6P8AAX8|
#### Variant Support
|Product | Promo code | 
|---------|------------|
|Digital |DJ6CCJZLL |
|Paper + Digital |NJ6CCSVMC |
|Paper|NJ6CCQLD4|
### Variant:Control

| Channel | Screenshot|
|---------|-----------------|
| **Engagement Banner** | picture 371 |
| **Epic** | picture 366 |
| **Live blog**  | picture 381  |

### Variant: Support

| Channel | Screenshot|
|---------|-----------------|
| **Engagement Banner** | picture 364 |
| **Epic** | picture 359 |
| **Live blog** | picture 380 |

## What is the value of this and can you measure success?
N/A

## Does this affect other platforms - Amp, Apps, etc?
N/A

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

No

## Tested in CODE?
Not Yet

**Support Variant** correctly Tested in **Code**

### Screen captures

Part of the site |  Support Variant |
|---------------|-----------------|
| Front page  |controlfrontpage|  
| Article |articlecontrol | 
|Liveblog |liveblogcontrol |

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: This could really be an inlined template literal. We could ditch the `lodash/utilities/template` then :)

**Comment by joelochlann** [comment]
Would be good to get some feedback from @jranks123 and the rest of @guardian/contributions

File: common/app/conf/switches/ABTestSwitches.scala
Comment (joelochlann) [inline common/app/conf/switches/ABTestSwitches.scala]: Whoops! Duplicate switch

**Comment by paperboyo** [comment]
For consistency with the rest of the header, I don’t think “Support the Guardian” should be capitalised?

File: common/app/conf/switches/ABTestSwitches.scala
Comment (svillafe) [inline common/app/conf/switches/ABTestSwitches.scala]: Fix in 630f206

**Comment by svillafe** [comment]
@paperboyo fixed in: ae5622d1e7dbb22002974713795d0810d4742a6f

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: I guess `<%=componentName%>` is a lodash leftover?

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Links without targets should be `<button>`s.

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Yeah - we can just delete that whole attribute, we don't use it for anything

**Review by gu-stav — APPROVED** [review]
The return type `string` is missing for `buildButtonTemplate()` and `liveblogEpicTemplate()`, but all the rest of the code looks good to me now.

If you have some time, it would be great, if `<a href="#" itemprop="url" class="block-time__link">` could become a `<button>` element. I leave this up to you, since it involes some extra styles I suppose.

**Comment by joelochlann** [comment]
@gustavpursche It is actually a link and not a button. It's just that `#` was a placeholder for a value I never filled in. I've fixed this in https://github.com/guardian/frontend/pull/17678

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: I guess we can remove this TODO

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: can get rid of this TODO

**Comment by joelochlann** [comment]
One note about the tracking on this one: going by our normal method of counting impressions, the conversion rate for both of these will be lower than normal because we will over-count impressions.

Why? Because we track impressions by registering an A/B test participation on a page view. For, say, an epic test, we won't register a participation if conditions for displaying the epic aren't met (e.g. the view count has been exceeded, or they're on a front rather than an article). But in this test, we have to bypass all of those checks in the `canRun` and replicate them in the test function.

So we would end up counting pretty much every single pageview by someone in the test as an impression, regardless of whether they saw the banner or the epic.

This might not be a problem since this will be the same in both variants and this test is a comparison.

But we also have component events to help mitigate this.

**Review by dominickendrick — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Why don't you use the same classes as above?

Comment (svillafe) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: I feel it is safer to have a new class name, just in case something else is using the `js-become-member` also.

Comment (NataliaLKB) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: In that case, I think a class should be added like `js-change-member-link` and `js-change-subscribe-link`. 

The principle behind that is the class should describe what its doing so there shouldn't be any surprises.

Comment (svillafe) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Cool, I will do that.

**Comment by NataliaLKB** [comment]
For the header work: 👍 ! 

Well done! 🎉

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Can we pass through the REFPVID here? For contributions too

Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: You can look at this function in contributions-utilities:

`
const addTrackingCodesToUrl = (base: string, campaignCode: string) => {
    const params = {
        REFPVID: (config.ophan && config.ophan.pageViewId) || 'not_found',
        INTCMP: campaignCode,
    };

    return `${base}?${constructURLQuery(params)}`;
};
`

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Was it necessary to duplicate the logic found in acquisition test selector?

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js
Comment (svillafe) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-baseline.js]: Added REFPVID in 877e3df3a8c5fddf9fe1bdafb5d4cd2c4352a283

**Review by jranks123 — APPROVED** [review]
Before we merge, I would reccomend double testing every link on every channel to make sure it is sending through the campaign codes as expected

**Comment by svillafe** [comment]
**Support** correctly Tested in **Code**

### Screen captures

Part of the site |  Support Variant |
|---------------|-----------------|
| Front page  |  frontpage |  
| Article | supportvariant| 
|Liveblog | liveblogsupport|

**Comment by svillafe** [comment]
**Control Variant** and **not in test** correctly Tested in **Code**

### Screen captures for Control

Part of the site |  Control Variant |
|---------------|-----------------|
| Front page  |controlfrontpage|  
| Article |articlecontrol | 
|Liveblog |liveblogcontrol |

**Review by joelochlann — APPROVED** [review]

**Pull request merged**
